src/core/document.js

```diff
@@ -715,6 +697,61 @@ class PDFDocument {
     return shadow(this, "numPages", num);
   }
 
+  /**
+   * @private
+   */
+  _hasOnlyDocumentSignatures(fields) {
+    return fields.every(field => {
+      field = this.xref.fetchIfRef(field);
+      if (field.has("Kids")) {
+        return this._hasOnlyDocumentSignatures(field.get("Kids"));
```

Comment:
Given that this is a *recursive* function call, you really must protect against infinite loops in this method!

One option could be to track all seen `Kids` nodes and break if anyone is ever repeated, however that relies on the assumption that they're always `Ref`s which may not be guaranteed in the general case.
The easiest solution is probably to just keep track of the recursion-depth, and break when it reaches some reasonable limit (e.g. `10`?).